### PR TITLE
[CPDLP-2513] NPQ Maths - Add new field to NPQ contracts

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

- Ticket: https://dfedigital.atlassian.net/browse/CPDLP-2513

### Changes proposed in this pull request

* Added new column `special_course` to `NPQContract`
* Updated seed files to enable `special_course` for Maths courses
* Created `Oneoffs::NPQ::SetSpecialCourseForNPQContracts` to allow setting `special_course` value for NPQ contracts.

### CODE REVIEW

**Review by cwrw — COMMENTED** [review]
looking good some minor comments

File: db/data/npq_contracts/fake-2022.csv [hunk 1]
```diff
@@ -1,87 +1,87 @@
...
-National Institute of Teaching,2022,npq-leading-behaviour-culture,101,801,24
-National Institute of Teaching,2022,npq-leading-teaching-development,102,802,24
-National Institute of Teaching,2022,npq-senior-leadership,103,803,24
-National Institute of Teaching,2022,npq-headship,104,804,24
-National Institute of Teaching,2022,npq-executive-leadership,105,805,24
-National Institute of Teaching,2022,npq-leading-literacy,106,806,24
-National Institute of Teaching,2022,npq-early-years-leadership,107,807,24
-National Institute of Teaching,2022,npq-early-headship-coaching-offer,1000,800,0
+provider_name,cohort_year,course_identifier,recruitment_target,per_participant,service_fee_installments,special_course
+Ambition Institute,2022,npq-leading-teaching,1000,100,12,
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply db/data/npq_contracts/fake-2022.csv:2]: Updated

File: db/data/npq_contracts/fake-2021.csv [hunk 2]
```diff
@@ -1,78 +1,78 @@
...
-Best Practice Network,2021,npq-early-headship-coaching-offer,300,800,0
-Church of England,2021,npq-early-headship-coaching-offer,300,800,0
-Education Development Trust,2021,npq-early-headship-coaching-offer,300,800,0
-School-Led Network,2021,npq-early-headship-coaching-offer,300,800,0
-LLSE,2021,npq-early-headship-coaching-offer,300,800,0
-Teach First,2021,npq-early-headship-coaching-offer,300,800,0
-Teacher Development Trust,2021,npq-early-headship-coaching-offer,300,800,0
-UCL Institute of Education,2021,npq-early-headship-coaching-offer,300,800,0
+provider_name,cohort_year,course_identifier,recruitment_target,per_participant,service_fee_installments,special_course
+Ambition Institute,2021,npq-leading-teaching,1000,900,12,
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply db/data/npq_contracts/fake-2021.csv:2]: Updated

File: db/schema.rb [hunk 3]
```diff
@@ -1239,7 +1240,7 @@
               count(*) AS count
              FROM (participant_profiles participant_profiles_1
                JOIN participant_identities participant_identities_1 ON ((participant_identities_1.id = participant_profiles_1.participant_identity_id)))
-            WHERE ((participant_profiles_1.type)::text = ANY (ARRAY[('ParticipantProfile::ECT'::character varying)::text, ('ParticipantProfile::Mentor'::character varying)::text]))
+            WHERE ((participant_profiles_1.type)::text = ANY ((ARRAY['ParticipantProfile::ECT'::character varying, 'ParticipantProfile::Mentor'::character varying])::text[]))
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply db/schema.rb:1243]: Updated

**Review by cwrw — APPROVED** [review]

**Pull request merged**